File: java/parser/src/main/java/gov/lbl/parser/parser/Element_modification_or_replaceableVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Element_modification;
import gov.lbl.parser.domain.Element_modification_or_replaceable;
import gov.lbl.parser.domain.Element_replaceable;

public class Element_modification_or_replaceableVisitor  extends modelicaBaseVisitor<Element_modification_or_replaceable> {
    @Override
    public Element_modification_or_replaceable visitElement_modification_or_replaceable(modelicaParser.Element_modification_or_replaceableContext ctx) {

        Boolean each = ctx.EACH() == null ? false : true;
        Boolean is_final = ctx.FINAL() == null ? false : true;

        Element_modificationVisitor element_modificationVisitor = new Element_modificationVisitor();
        Element_modification element_modification = ctx.element_modification() == null ? null : ctx.element_modification().accept(element_modificationVisitor);

        Element_replaceableVisitor element_replaceableVisitor = new Element_replaceableVisitor();
        Element_replaceable element_replaceable = ctx.element_replaceable() == null ? null : ctx.element_replaceable().accept(element_replaceableVisitor);
        
        return new Element_modification_or_replaceable(each, is_final, element_modification, element_replaceable);
    }
}

File: jsParser/parser/Element_modification_or_replaceableVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Element_modification_or_replaceable = require('../domain/Element_modification_or_replaceable')

const Element_modificationVisitor = require('./Element_modificationVisitor')
const Element_replaceableVisitor = require('./Element_replaceableVisitor')

class Element_modification_or_replaceableVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitElement_modification_or_replaceable (ctx) {
    const each = !!ctx.EACH()
    const is_final = !!ctx.FINAL()
    let element_modification = null
    let element_replaceable = null

    if (ctx.element_modification()) {
      const element_modificationVisitor = new Element_modificationVisitor.Element_modificationVisitor()
      element_modification = element_modificationVisitor.visitElement_modification(ctx.element_modification())
    }
    if (ctx.element_replaceable()) {
      const element_replaceableVisitor = new Element_replaceableVisitor.Element_replaceableVisitor()
      element_replaceable = element_replaceableVisitor.visitElement_replaceable(ctx.element_replaceable())
    }

    return new Element_modification_or_replaceable.Element_modification_or_replaceable(each, is_final, element_modification, element_replaceable)
  }
};

Element_modification_or_replaceableVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitElement_modification_or_replaceable = this.visitElement_modification_or_replaceable
exports.Element_modification_or_replaceableVisitor = Element_modification_or_replaceableVisitor
